packages/material-ui-lab/src/Slider/Slider.js

```diff
@@ -0,0 +1,523 @@
+    secondary: '#bdbdbd',
+    focused: '#9e9e9e',
+    disabled: '#bdbdbd',
+  };
+
+  return {
+    /* Styles for wrapper container */
+    container: {
+      position: 'relative',
+      width: '100%',
+      padding: '16px 0',
+      '&$reverse': {
+        transform: 'scaleX(-1)',
+      },
+      '&$vertical': {
+        position: 'relative',
+        height: '100%',
+        padding: '0 16px',
+        '&$reverse': {
+          transform: 'scaleY(-1)',
+        },
+      },
+    },
+    /* Tracks styles */
+    track: {
+      position: 'absolute',
+      transform: 'translate(0, -50%)',
+      top: '50%',
+      height: 2,
+      '&$vertical': {
+        transform: 'translate(-50%, 0)',
+        left: '50%',
+        top: 'initial',
+        width: 2,
+      },
+      '&$disabled': {
+        backgroundColor: colors.secondary,
+      },
+    },
+    trackBefore: {
+      zIndex: 1,
+      left: 0,
+      backgroundColor: colors.primary,
+      transition: commonTransitions,
+      '&$vertical': {
+        top: 0,
+      },
+    },
+    trackAfter: {
+      right: 0,
+      backgroundColor: colors.secondary,
+      transition: commonTransitions,
+      '&$vertical': {
+        bottom: 0,
+      },
+      '&$active': {
+        backgroundColor: colors.focused,
+      },
+    },
+    /* Thumb styles  */
+    thumb: {
+      position: 'absolute',
+      zIndex: 2,
+      transform: 'translate(-50%, -50%)',
+      width: 12,
+      height: 12,
+      borderRadius: '50%',
+      transition: commonTransitions,
+      backgroundColor: colors.primary,
+      '&:focus': {
+        boxShadow: `0px 0px 0px 10px ${fade(colors.primary, 0.3)}`,
+      },
+      '&$active': {
+        width: 18,
+        height: 18,
+        '&:focus': {
+          boxShadow: 'none',
+        },
+        '&$initial': {
+          border: `2px solid ${colors.focused}`,
+        },
+      },
+      '&$disabled': {
+        width: 10,
+        height: 10,
+        cursor: 'no-drop',
+        backgroundColor: colors.disabled,
+        '&:focus': {
+          border: `2px solid ${colors.disabled}`,
+          boxShadow: 'none',
+        },
+      },
+      '&$initial': {
+        border: `2px solid ${colors.disabled}`,
+        backgroundColor: 'transparent',
+        '&:focus': {
+          border: `2px solid ${colors.focused}`,
+          boxShadow: `0px 0px 0px 10px ${fade(colors.disabled, 0.3)}`,
+        },
+      },
+    },
+    initial: {},
+    vertical: {},
+    disabled: {},
+    focus: {},
+    active: {},
+    reverse: {},
+  };
+};
+
+const KEY_CODES = {
+  ARROW_LEFT: 37,
+  ARROW_UP: 38,
+  ARROW_RIGHT: 39,
+  ARROW_DOWN: 40,
+  HOME: 36,
+  END: 35,
+};
+
+function addEventListener(node, event, handler, capture) {
+  node.addEventListener(event, handler, capture);
+  return {
+    remove: function remove() {
+      node.removeEventListener(event, handler, capture);
+    },
+  };
+}
+
+function normalizeValue(value, min = 0, max = 100) {
```

Comment:
We call this `clamp` elsewhere (CircularProgress and colorManipulator, where it also gives a warning for out-of-range values. I wonder if we should move it to utils as a shared function?

Reply:
Looking at the implementation of this vs CircularProgress, that seems reasonable.

Reply:
Where I should place function? In /src/utils or in /packages/material-ui-lab/src/utils ?